hack/generator/pkg/codegen/pipeline/report_resource_versions.go

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/config"
+)
+
+// ReportResourceVersionsStageID is the unique identifier of this stage
+const ReportResourceVersionsStageID = "reportResourceVersions"
+
+// ReportResourceVersions creates a pipeline stage that removes any wrapper types prior to actual code generation
+func ReportResourceVersions(configuration *config.Configuration) Stage {
+
+	return MakeStage(
+		ReportResourceVersionsStageID,
+		"Generate a report listing all the resources generated",
+		func(ctx context.Context, state *State) (*State, error) {
+			report := NewResourceVersionsReport()
+			report.Summarize(state.Types())
+			err := report.WriteTo(configuration.FullTypesOutputPath())
+			return state, err
+		})
+}
+
+type ResourceVersionsReport struct {
+	// A separate list of resources for each package
+	lists map[astmodel.PackageReference][]string
+}
+
+func NewResourceVersionsReport() *ResourceVersionsReport {
+	return &ResourceVersionsReport{
+		lists: make(map[astmodel.PackageReference][]string),
+	}
+}
+
+// Summarize collates a list of all resources, grouped by package
+func (r *ResourceVersionsReport) Summarize(types astmodel.Types) {
+	resources := astmodel.FindResourceTypes(types)
+	for _, rsrc := range resources {
+		name := rsrc.Name()
+		pkg := name.PackageReference
+		r.lists[pkg] = append(r.lists[pkg], name.Name())
+	}
+}
+
+func (r *ResourceVersionsReport) WriteTo(outputPath string) error {
+
```

Comment:
very minor: You have a newline here but not on the other methods

Reply:
Fixed.